### PR TITLE
updating to the latest cloudevents sdk - v2.0.0-RC1

### PR DESCRIPTION
**Pull request by n3wscott**
Fixes https://github.com/knative/docs/issues/2382

## Proposed Changes 

- Update two samples to latest CloudEvents Go SDK version. New version is v2 in rc1.

/cc slinkydeveloper

### CODE REVIEW

File: docs/eventing/samples/helloworld/helloworld-go/README.md [hunk 1]
```diff
@@ -24,65 +24,46 @@ cd knative-docs/docs/eventing/samples/helloworld/helloworld-go
...
-      "fmt"
-      "log"
-      "net/http"
-      "os"
-
-      cloudevents "github.com/cloudevents/sdk-go"
-      "github.com/google/uuid"
+        "context"
+        "log"
+
```
Comment (mattmoor) [inline docs/eventing/samples/helloworld/helloworld-go/README.md:30]: Remove trailing whitespace:
```suggestion

```

File: docs/eventing/samples/helloworld/helloworld-go/README.md [hunk 2]
```diff
@@ -24,65 +24,46 @@ cd knative-docs/docs/eventing/samples/helloworld/helloworld-go
...
-    }
-
-    func handler(w http.ResponseWriter, r *http.Request) {
-      log.Print("Hello world received a request.")
-      target := os.Getenv("TARGET")
-      if target == "" {
-        target = "World"
-      }
-      fmt.Fprintf(w, "Hello %s!\n", target)
+
```
Comment (mattmoor) [inline docs/eventing/samples/helloworld/helloworld-go/README.md:34]: Remove trailing whitespace:
```suggestion

```

File: docs/eventing/samples/helloworld/helloworld-go/README.md [hunk 3]
```diff
@@ -24,65 +24,46 @@ cd knative-docs/docs/eventing/samples/helloworld/helloworld-go
...
+        // Here is where your code to process the event will go.
+        // In this example we will log the event msg
+        log.Printf("Event received. \n%s\n", event)
+        data := &HelloWorld{}
+        if err := event.DataAs(data); err != nil {
+            log.Printf("Error while extracting cloudevent Data: %s\n", err.Error())
+            return nil, cloudevents.NewHTTPResult(400, "failed to convert data: %s", err)
+        }
+        log.Printf("Hello World Message from received event %q", data.Msg)
+
```
Comment (mattmoor) [inline docs/eventing/samples/helloworld/helloworld-go/README.md:45]: Remove trailing whitespace:
```suggestion

```

File: docs/eventing/samples/helloworld/helloworld-go/README.md [hunk 4]
```diff
@@ -24,65 +24,46 @@ cd knative-docs/docs/eventing/samples/helloworld/helloworld-go
...
+        newEvent.SetSource("knative/eventing/samples/hello-world")
+        newEvent.SetType("dev.knative.samples.hifromknative")
+        if err := newEvent.SetData(cloudevents.ApplicationJSON, HiFromKnative{Msg: "Hi from helloworld-go app!"}); err != nil {
+            return nil, cloudevents.NewHTTPResult(500, "failed to set response data: %s", err)
+        }
+        log.Printf("Responding with event\n%s\n", newEvent)
+        return &newEvent, nil
     }
-
+
```
Comment (mattmoor) [inline docs/eventing/samples/helloworld/helloworld-go/README.md:59]: Remove trailing whitespace:
```suggestion

```

File: docs/eventing/samples/helloworld/helloworld-go/README.md [hunk 5]
```diff
@@ -271,8 +252,10 @@ Helloworld-go app logs the context and the msg of the above event, and replies b
       ```
       You should see something similar to:
       ```shell
-      Event received. Context: Context Attributes,
-        specversion: 0.3
+      Event received.
```
Comment (mattmoor) [inline docs/eventing/samples/helloworld/helloworld-go/README.md:255]: Remove trailing whitespace:
```suggestion
      Event received.
```

File: docs/eventing/samples/helloworld/helloworld-go/README.md [hunk 6]
```diff
@@ -282,14 +265,18 @@ Helloworld-go app logs the context and the msg of the above event, and replies b
         knativearrivaltime: 2019-10-04T22:35:26Z
         knativehistory: default-kn2-trigger-kn-channel.knative-samples.svc.cluster.local
         traceparent: 00-971d4644229653483d38c46e92a959c7-92c66312e4bb39be-00
-
+      Data,
+        {"msg":"Hello World from the curl pod."}
+
```
Comment (mattmoor) [inline docs/eventing/samples/helloworld/helloworld-go/README.md:270]: Remove trailing whitespace:
```suggestion

```

File: docs/serving/samples/cloudevents/cloudevents-go/cloudevents.go [hunk 7]
```diff
@@ -55,16 +54,15 @@ type Response struct {
 }
 
 // handle shared the logic for producing the Response event from the Request.
-func handle(req Request) (resp Response) {
-	resp.Message = fmt.Sprintf("Hello, %s", req.Name)
-	return
+func handle(req Request) Response {
+	return Response{Message:fmt.Sprintf("Hello, %s", req.Name)}
```
Comment (mattmoor) [inline docs/serving/samples/cloudevents/cloudevents-go/cloudevents.go:58]: Format Go code:
```suggestion
	return Response{Message: fmt.Sprintf("Hello, %s", req.Name)}
```

**Comment by n3wscott** [comment]
> Not for this PR, but switching this to use `go mod` (without commiting the vendored code) within each sample directory seems like it would be a win.

totally.

**Pull request merged**